# Find the mean of old scheme and new scheme column. (5 points)


In [1]:
import numpy as np
import pandas as pd

In [7]:
titan_sales_df=pd.read_excel(io="Titan_Data.xls")
titan_sales_df['Old Scheme (in thousands)'].mean()
titan_sales_df['New Scheme (in thousands)'].mean()
print("Mean of Old Scheme is {}".format(titan_sales_df['Old Scheme (in thousands)'].mean()))
print("Mean of New Scheme is {}".format(titan_sales_df['New Scheme (in thousands)'].mean()))




Mean of Old Scheme is 68.03333333333333
Mean of New Scheme is 72.03333333333333


# Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? (10 points)


In [8]:
from scipy.stats import ttest_rel
ttest_rel(titan_sales_df['Old Scheme (in thousands)'],titan_sales_df['New Scheme (in thousands)'])


Ttest_relResult(statistic=-1.5559143823544377, pvalue=0.13057553961337662)

# What conclusion does the test (p-value) lead to? (2.5 points)

#Null Hypothesis : There is no significant Increase in Output Sales. i.e there is no differance between Mean before New Sales Policy and Mean After Sales policy.  ( Population Mean Before Sales Policy = Population Mean after Sales Policy)

#Alternate Hypothesis : New Scheme has significantly Raised Output Sales. i.e Population Mean After Sales Policy > Population Mean Before Sales Polic1y ( Only Upper Tail). Lower Tail will be the Population Mean After Sales Policy < Population Mean Before Sales Policy.

#Since t(static) is not less than Critical T( alpha=.05/2) ( -1.5559 not less than -2.0452) or t(static) is not greater than Critical T( alpha=.05)/2) ( -1.5559 not greater than 2.0452). So We accept Null Hypothesis.

# So New Scheme has not increased the Sales Output

# 4. Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:


#        a) The probability of a type 1 error? (2.5 points)



In [12]:
#Null Hypothesis: The Differance in Sales Mean is equal 5 (in thousands)
#Alternate Hypothesis: The Differance in Sales Mean is Not Equal to  5 ( 2 Tailed)
#Alternate Hypothesis: The Differance in Sales Mean is Greater than  5 ( Upper Tailed)
#Alternate Hypothesis: The Differance in Sales Mean is Less than  5 ( Lower Tailed)
# Caluclating the test for  Population Mean  using Single T Test.

import scipy.stats as stas
titan_diff_df=titan_sales_df['New Scheme (in thousands)'] - titan_sales_df['Old Scheme (in thousands)']
t_stat,p_value=stas.ttest_1samp(a=titan_diff_df,popmean=5)
print("T_static Value is {} and Probablilty is {}".format(t_stat,p_value))

T_static Value is -0.3889785955886094 and Probablilty is 0.7001334912613286


In [17]:
# So if the average output must increase by £5000 in the scheme compared to the old scheme. 
#( i.e New Scheme Mean must be greater than or equal to 73.034). So 

type1_error= p_value + (1 - p_value)/2

print("The Type 1 Error for the Differance in Sales Mean >= 5 is {}".format(type1_error))

The Type 1 Error for the Differance in Sales Mean >= 5 is 0.8500667456306643


# b) What is the p- value of the hypothesis test if we test for a difference of 5000? (10 points)



In [18]:
# So if the average output must increase by £5000 in the scheme compared to the old scheme. 
#( i.e New Scheme Mean must be greater than or equal to 73.034). So 

print("The probability for the Differance in Sales Mean >= 5 is {}".format(type1_error))

The probability for the Differance in Sales Mean >= 5 is 0.8500667456306643


#        c) Power of the test (5 points)



In [19]:
# Probablilty of Null Hypothesis is False is 
1-p_value
print("The probability for the Null Hypothesis being False. i.e Beta is {}".format(1-p_value))

The probability for the Null Hypothesis being False. i.e Beta is 0.2998665087386714


In [20]:
print("The Power of Test is {}".format(1-(1-p_value)))

The Power of Test is 0.7001334912613286
